echopype/tests/commongrid/test_api.py

```diff
@@ -0,0 +1,256 @@
+            3,
+            10,
+        )  # full array, no NaN
+        assert (ds_MVBS["Sv"].isel(ping_time=slice(3, 12)).dropna(dim="echo_range").shape) == (
+            2,
+            9,
+            7,
+        )  # bottom bins contain NaN
+        assert (ds_MVBS["Sv"].isel(ping_time=slice(12, None)).dropna(dim="echo_range").shape) == (
+            2,
+            6,
+            3,
+        )  # bottom bins contain NaN
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    ("er_type"),
+    [
+        ("regular"),
+        ("irregular"),
+    ],
+)
+def test_compute_MVBS_values(request, er_type):
+    """Tests for the values of compute_MVBS on regular and irregular data."""
+
+    def _parse_nans(mvbs, ds_Sv) -> np.ndarray:
+        """Go through and figure out nan values in result"""
+        echo_range_step = np.unique(np.diff(mvbs.Sv.echo_range.values))[0]
+        expected_outs = []
+        # Loop over channels
+        for chan in mvbs.Sv.channel.values:
+            # Get ping times for this channel
+            ping_times = mvbs.Sv.ping_time.values
+            # Compute the total number of pings
+            ping_len = len(ping_times)
+            # Variable to store the expected output for this channel
+            chan_expected = []
+            for idx in range(ping_len):
+                # Loop over pings and create slices
+                if idx < ping_len - 1:
+                    ping_slice = slice(ping_times[idx], ping_times[idx + 1])
+                else:
+                    ping_slice = slice(ping_times[idx], None)
+
+                # Get the original echo_range data for this channel and ping slice
+                da = ds_Sv.echo_range.sel(channel=chan, ping_time=ping_slice)
+                # Drop the nan values since this shouldn't be included in actual
+                # computation for compute_MVBS, a.k.a. 'nanmean'
+                mean_values = da.dropna(dim="ping_time", how="all").values
+                # Compute the histogram of the mean values to get distribution
+                hist, _ = np.histogram(
+                    mean_values[~np.isnan(mean_values)],
+                    bins=np.append(
+                        mvbs.Sv.echo_range.values,
+                        # Add one more bin to account for the last value
+                        mvbs.Sv.echo_range.values.max() + echo_range_step
+                    ),
+                )
+                # Convert any non-zero values to False, and zero values to True
+                # to imitate having nan values since there's no value for that bin
+                chan_expected.append([False if v > 0 else True for v in hist])
+            expected_outs.append(chan_expected)
+        return np.array(expected_outs)
+
+    range_bin = "2m"
+    ping_time_bin = "1s"
+
+    if er_type == "regular":
+        ds_Sv = request.getfixturevalue("mock_sv_dataset_regular")
+        expected_mvbs = request.getfixturevalue("mock_mvbs_array_regular")
```

Comment:
I noticed that the values to be tested against are hard-coded. I think it should be computed on the fly using a brute-force loop for the most accurate verification, sine the dataset is tiny.

I did that, and the results are different. If you compute `expected_mvbs_val` using the code below, you'll notice that it does not correspond to the values given by flox. I chased down the cause of it, and it is that along the `echo_range` axis: for the first bin (range between 0 and 2), flox seems to slice out indices that correspond to `echo_range` values (0.5, 1, 1.5, 2), instead of (0, 0.5, 1, 1.5).

Could you look into what's going on, and change the test to be against brute-force looping outputs? Thanks!
```python
    # create bin information needed for ping_time
    d_index = (
        ds_Sv["ping_time"]
        .resample(ping_time=ping_time_bin, skipna=True)
        .first()  # Not actually being used, but needed to get the bin groups
        .indexes["ping_time"]
    )
    ping_interval = d_index.union([d_index[-1] + pd.Timedelta(ping_time_bin)]).values

    # create bin information for echo_range
    # this computes the echo range max since there might NaNs in the data
    echo_range_max = ds_Sv["echo_range"].max()
    range_interval = np.arange(0, echo_range_max + 2, 2)

    sv = ep.utils.compute._log2lin(ds_Sv["Sv"].data)

    expected_mvbs_val = np.ones((2, len(ping_interval)-1, len(range_interval)-1)) * np.nan
    for ch_idx in range(2):
        for p_idx in range(len(ping_interval)-1):
            for r_idx in range(len(range_interval)-1):
                p_idx_active = np.logical_and(
                    ds_Sv["ping_time"].data >= ping_interval[p_idx],
                    ds_Sv["ping_time"].data < ping_interval[p_idx+1]
                )
                r_idx_active = np.logical_and(
                    ds_Sv["echo_range"].isel(channel=0, ping_time=0).data >= range_interval[r_idx],
                    ds_Sv["echo_range"].isel(channel=0, ping_time=0).data < range_interval[r_idx+1]
                )
                # print(p_idx_active)
                # print(r_idx_active)
                sv_tmp = sv.data[ch_idx][np.ix_(p_idx_active, r_idx_active)]
                expected_mvbs_val[ch_idx, p_idx, r_idx] = sv_tmp.mean()
    expected_mvbs_val = ep.utils.compute._lin2log(expected_mvbs_val)
```